From e680d71b75f18be9b99b13a3a2f63e5f54e42d3f Mon Sep 17 00:00:00 2001
From: Michael Stahl <mstahl@redhat.com>
Date: Sat, 24 Aug 2013 01:19:21 +0200
Subject: [PATCH] fdo#68319: sw: fix Chinese Conversion

... and perhaps other things too: the SwScriptIterator constructor gets
passed a temporary String created by implicit conversion from OUString
and retains a reference to that; don't convert it to OUString like on
master but just cast to String.

(regression from 0295c8a34e39326414c1b98cf4da905802f061b0)

Change-Id: I30c7309c14ec8f9e41ee5fe89eed513ddad154fe
Reviewed-on: https://gerrit.libreoffice.org/5600
Reviewed-by: Fridrich Strba <fridrich@documentfoundation.org>
Tested-by: Fridrich Strba <fridrich@documentfoundation.org>
---
 sw/source/core/txtnode/txatritr.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sw/source/core/txtnode/txatritr.cxx b/sw/source/core/txtnode/txatritr.cxx
index 00e007a..b9a15b6 100644
--- a/sw/source/core/txtnode/txatritr.cxx
+++ b/sw/source/core/txtnode/txatritr.cxx
@@ -94,7 +94,7 @@ bool SwScriptIterator::Next()
 SwTxtAttrIterator::SwTxtAttrIterator( const SwTxtNode& rTNd, sal_uInt16 nWhchId,
                                         xub_StrLen nStt,
                                         bool bUseGetWhichOfScript )
-    : aSIter( rTNd.GetTxt(), nStt ), rTxtNd( rTNd ),
+    : aSIter( reinterpret_cast<const String&>(rTNd.GetTxt()), nStt ), rTxtNd( rTNd ),
     pParaItem( 0 ), nChgPos( nStt ), nAttrPos( 0 ), nWhichId( nWhchId ),
     bIsUseGetWhichOfScript( bUseGetWhichOfScript )
 {
-- 
1.8.3.1

